sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -131,12 +131,23 @@ def get_historical_features(
             feature_refs, feature_views, registry, project
         )
 
-        # TODO: Infer min_timestamp and max_timestamp from entity_df
+        # Infer min_timestamp and max_timestamp from entity_df
+        if isinstance(entity_df, pandas.DataFrame):
+            min_timestamp = datetime.fromisoformat(
+                min(entity_df[[DEFAULT_ENTITY_DF_EVENT_TIMESTAMP_COL]])
+            )
+            max_timestamp = datetime.fromisoformat(
+                max(entity_df[DEFAULT_ENTITY_DF_EVENT_TIMESTAMP_COL])
+            )
+        else:
+            min_timestamp = datetime.now() - timedelta(days=365)
+            max_timestamp = datetime.now() + timedelta(days=1)
```

Comment:
In the case that `entity_df` is a `str` (a query that gets run in BQ), I think `_upload_entity_df_into_bigquery` a few lines [above](https://github.com/feast-dev/feast/blob/master/sdk/python/feast/infra/offline_stores/bigquery.py#L114-L120) will create a tem table that expires in 30 minutes. (see definition [here](https://github.com/feast-dev/feast/blob/master/sdk/python/feast/infra/offline_stores/bigquery.py#L335-L338))

I'm not sure how clean it'd be to try to extract mins and maxes from that table vs just having some kind of defaults for min and max timestamp in this case, but could be something to look into. Hopefully others have thoughts too.

Reply:
Thanks Cody, in the latest change I infer and use the timestamps purely in SQL query (lines 464-469) so this is will also address the issue you point out!